### ISSUE TITLE
Shaded jar contains old incompatible version of reactor-core

### ISSUE THREAD

**Issue by dsyer** [cloudfoundry/java-buildpack-auto-reconfiguration#73]
If you build an app with webflux and Spring Boot 2.2 and then add this jar to the classpath it barfs on every HTTP request because of an old version of reactor-core that gets shaded into the auto-reconfiguration jar:

```
2019-10-16 11:05:26.798 ERROR 118 --- [or-http-epoll-2] reactor.netty.http.server.HttpServer     : [id: 0x87411ee4, L:/172.17.0.2:8080 - R:/172.17.0.1:39434] 

java.lang.NoSuchMethodError: 'reactor.core.publisher.Mono reactor.core.publisher.Mono.transformDeferred(java.util.function.Function)'
	at org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter.filter(MetricsWebFilter.java:88) ~[spring-boot-actuator-2.2.0.RC1.jar:2.2.0.RC1]
	at org.springframework.web.server.handler.DefaultWebFilterChain.invokeFilter(DefaultWebFilterChain.java:126) ~[spring-web-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.web.server.handler.DefaultWebFilterChain.lambda$filter$0(DefaultWebFilterChain.java:121) ~[spring-web-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44) ~[java-buildpack-auto-reconfiguration-2.10.0.RELEASE.jar:na]
	...
```

**Comment by dsyer** [comment]
It looks like maybe the reactor dependency is pulled in via spring-data-redis. It's fixed in Spring Boot 2.2, but with the 2.1 version used here it pulls in reactor-core and reactive-streams with scope=compile (so they are added to the shaded jar inadvertently and unnecessarily).

**Comment by dsyer** [comment]
The problem is with reactor (https://github.com/reactor/reactor/issues/677). To fix it here we would need to declare the test scope explicitly for all the reactor dependencies.
